components/cache/index.js

```diff
@@ -0,0 +1,427 @@
+					if (err) {
+						return whilstCb(err);
+					}
+					cursor = Number(res.shift());
+					keys = res.shift();
+					if (keys.length > 0) {
+						return this.client.del(keys, whilstCb);
+					}
+					return whilstCb();
+				});
+			},
+			() => cursor > 0,
+			cb
+		);
+	}
+
+	/**
+	 * Removes all entries from redis db.
+	 *
+	 * @param {function} cb
+	 * @todo Add description for the params
+	 * @todo Add @returns tag
+	 */
+	flushDb(cb) {
+		this.logger.debug('Cache - Flush database');
+		if (!this.isConnected()) {
+			return cb(errorCacheDisabled);
+		}
+		return this.client.flushdb(cb);
+	}
+
+	/**
+	 * Quits established redis connection upon process exit.
+	 *
+	 * @param {function} cb
+	 * @todo Add description for the params
+	 * @todo Add @returns tag
+	 */
+	cleanup(cb) {
+		this.logger.debug('Cache - Clean up database');
+		this.quit(cb);
+	}
+
+	/**
+	 * Quits established redis connection.
+	 *
+	 * @param {function} cb
+	 * @todo Add description for the params
+	 * @todo Add @returns tag
+	 */
+	quit(cb) {
+		this.logger.debug('Cache - Quit database');
+		if (!this.isConnected()) {
+			// Because connection is not established in the first place
+			return cb();
+		}
+		return this.client.quit(cb);
+	}
+
+	/**
+	 * Clears cache entries for given pattern.
+	 *
+	 * @param {string} pattern
+	 * @param {function} cb
+	 * @todo Add description for the params
+	 * @todo Add @returns tag
+	 */
+	clearCacheFor(pattern, cb) {
+		cb = cb || function() {};
+
+		this.logger.debug(
+			['Cache - clearCacheFor', pattern, '| Status:', this.isConnected()].join(
+				' '
+			)
+		);
+
+		if (!this.isReady()) {
+			return cb(errorCacheDisabled);
+		}
+
+		return this.removeByPattern(pattern, err => {
+			if (err) {
+				this.logger.error(
+					[
+						'Cache - Error clearing keys with pattern:',
+						pattern,
+						'on new block',
+					].join(' ')
+				);
+			} else {
+				this.logger.debug(
+					[
+						'Cache - Keys with pattern:',
+						pattern,
+						'cleared from cache on new block',
+					].join(' ')
+				);
+			}
+
+			cb(err);
+		});
+	}
+
+	/**
+	 * Clears all cache entries upon round finish.
+	 *
+	 * @param {Object} round - Current round.
+	 * @param {function} cb
+	 * @todo Add description for the params
+	 * @todo Add @returns tag
+	 */
+	onFinishRound(round, cb) {
+		cb = cb || function() {};
+
+		this.logger.debug(
+			['Cache - onFinishRound', '| Status:', this.isConnected()].join(' ')
+		);
+		if (!this.isReady()) {
+			return cb(errorCacheDisabled);
+		}
+		const pattern = CACHE.KEYS.delegatesApi;
+		return this.removeByPattern(pattern, err => {
+			if (err) {
+				this.logger.error(
+					[
+						'Cache - Error clearing keys with pattern:',
+						pattern,
+						'round finish',
+					].join(' ')
+				);
+			} else {
+				this.logger.debug(
+					[
+						'Cache - Keys with pattern:',
+						pattern,
+						'cleared from cache on new Round',
+					].join(' ')
+				);
+			}
+			return cb(err);
+		});
+	}
```

Comment:
This logic should be moved away from the shared component. Its strictly related to chain module and should come under round namespace in it.